src/Model/GroundWaterFlow/gwf3vsc8.f90

```diff
@@ -0,0 +1,1519 @@
+      this%cviscref(ispec) = input_data%cviscref(ispec)
+      this%cmodelname(ispec) = input_data%cmodelname(ispec)
+      this%cauxspeciesname(ispec) = input_data%cauxspeciesname(ispec)
+    end do
+
+  end subroutine set_packagedata
+
+  subroutine vsc_calcvisc(this)
+! ******************************************************************************
+! vsc_calcvisc -- calculate fluid viscosity from concentration
+! ******************************************************************************
+!
+!    SPECIFICATIONS:
+! ------------------------------------------------------------------------------
+    ! -- dummy
+    class(GwfVscType) :: this
+
+    ! -- local
+    integer(I4B) :: n
+    integer(I4B) :: i
+! ------------------------------------------------------------------------------
+    !
+    ! -- Calculate the viscosity using the specified concentration and/or
+    !    temperature arrays
+    do n = 1, this%dis%nodes
+      do i = 1, this%nviscspecies
+        if (this%modelconc(i)%icbund(n) == 0) then
+          this%ctemp = DZERO
+        else
+          this%ctemp(i) = this%modelconc(i)%conc(n)
+        end if
+      end do
+      !
+      this%visc(n) = calc_visc(this%ivisc, this%viscref, this%dviscdc, &
+                               this%cviscref, this%ctemp, this%a2, &
+                               this%a3, this%a4)
+
+    end do
+    !
+    ! -- Return
+    return
+  end subroutine vsc_calcvisc
+
+  subroutine allocate_scalars(this)
+! ******************************************************************************
+! allocate_scalars
+! ******************************************************************************
+!
+!    SPECIFICATIONS:
+! ------------------------------------------------------------------------------
+    ! -- modules
+    use ConstantsModule, only: DZERO
+    ! -- dummy
+    class(GwfVscType) :: this
+    ! -- local
+! ------------------------------------------------------------------------------
+    !
+    ! -- allocate scalars in NumericalPackageType
+    call this%NumericalPackageType%allocate_scalars()
+    !
+    ! -- Allocate
+    call mem_allocate(this%thermivisc, 'THERMIVISC', this%memoryPath)
+    call mem_allocate(this%idxtmpr, 'IDXTMPR', this%memoryPath)
+    call mem_allocate(this%ioutvisc, 'IOUTVISC', this%memoryPath)
+    call mem_allocate(this%ireadelev, 'IREADELEV', this%memoryPath)
+    call mem_allocate(this%iconcset, 'ICONCSET', this%memoryPath)
+    call mem_allocate(this%viscref, 'VISCREF', this%memoryPath)
+    call mem_allocate(this%a2, 'A2', this%memoryPath)
+    call mem_allocate(this%a3, 'A3', this%memoryPath)
+    call mem_allocate(this%a4, 'A4', this%memoryPath)
+    !
+    call mem_allocate(this%nviscspecies, 'NVISCSPECIES', this%memoryPath)
+    !
+    ! -- Initialize
+    this%thermivisc = 0
+    this%idxtmpr = 0
+    this%ioutvisc = 0
+    this%ireadelev = 0
+    this%iconcset = 0
+    this%viscref = 1000.d0
+    this%A2 = DZERO
+    this%A3 = DZERO
+    this%A4 = DZERO
```

Comment:
did I see earlier that these have default values?  shouldn't they go here?

Reply:
Great catch.  Defaults are not zero.  Have updated.